Methods rewrite part2 (#58)
greenelab/text_mined_hetnet_manuscript: content/03.methods.md

It integrates multiple signals emitted by label functions to assign each candidate sentence the most appropriate training class.
This model takes as input a label function output in the form of a matrix where rows represent candidate sentences, and columns represent each label function ($\Lambda^{nxm}$).
Once constructed, this model treats the true training class ($Y$) as a latent variable and assumes that each label function is independent of one another.
Under these two assumptions, the model finds the optimal parameters by minimizing a loglikelihood function marginalized over the latent training class.

$$
\hat{\theta} = argmin_{\theta}\sum_{Y}-log(P_{\theta}(\Lambda, Y)) 
$$

Following optimization, the model emits a probability estimate that each sentence belongs to the positive training class.
At this step, each probability estimate can be discretized via a chosen threshold into a positive or negative class.
We used a threshold of 0.5 for discretizing our training classes within our analysis.
For more information on how the likelihood function is constructed and minimized, refer to [@doi:10.1007/s00778-019-00552-1].

#### Discriminative Model

> The discriminative model is the ending step in this framework.

Reviewer: ```suggestion
The discriminative model is the final step in this framework.
```

This model uses training labels generated from the generative model combined with sentence features to classify the presence of a biomedical relationship.

> Typically, the discriminative model consists of some form of a neural network architecture to accomplish the above task.

Reviewer: ```suggestion
Typically, the discriminative model is a neural network.
```
